### ISSUE TITLE
Store additional dependencies in rockspec

### ISSUE THREAD

**Issue by pirogoeth** [keplerproject/luarocks#528]
It would be nice to be able to store additional deps, such as testing deps, inside the rockspec and provide a simple way to install them. For example, we'd have a structure like:
```
extra_dependencies = {
  test = {
    "busted >= 2.0.rc11-0",
    "luacov >= 0.10.0-1",
  },
  dev = {
    "remdebug >= 1.0-1",
    "luaprompt >= 0.7-1",
  }
}
...
```

And then, something like this for installing:
```
# If a local rockspec is present
luarocks install-deps dev test

# For a package pulled from remote
luarocks install --deps=dev,test <package>
```

Would something like this be possible?

**Comment by jdesgats** [comment]
Agreed, sometimes testing suite require a lot of dependencies. Make them optional could save a lot of pain sometimes, for instance when these dependencies are C modules, which themselves require extra libraries and their development headers.

**Comment by idchlife** [comment]
are there update on this issue? Is there possibility to a) save dependencies when installing --local to rockspec and installing them from rockspec?
